example/src/bin/readme.rs

```diff
@@ -7,11 +7,13 @@ const CACHE_NAME: &str = "cache";
 
 #[tokio::main]
 pub async fn main() -> Result<(), MomentoError> {
-    let cache_client = CacheClient::new(
-        CredentialProvider::from_env_var("MOMENTO_API_KEY".to_string())?,
-        laptop::latest(),
-        Duration::from_secs(60),
-    )?;
+    let cache_client = CacheClient::builder()
+        .default_ttl(Duration::from_secs(60))
+        .configuration(laptop::latest())
+        .credential_provider(CredentialProvider::from_env_var(
+            "MOMENTO_API_KEY".to_string(),
+        )?)
```

Comment:
this from_env_var function should take an `impl Into<String>` so that users do not need to manually convert `&'static str` into String. That said, I imagine you should just use an `&str` instead because I don't expect you have use of the environment variable after the function returns right?

Reply:
I have a different ticket about taking a pass over everything and cleaning up the String arg stuff, so I probably won't handle that in this PR, but - what would your general guidance be on this? Always prefer `&str` unless we need to take ownership? and if we need to take ownership then always prefer `impl Into<String>`?

Reply:
* In tests, always take `impl Into<String>` because that gives you the most flexibility and you do not care about memcpy cost
* In setup code _probably_ take `impl Into<String>` because that bothers users as little as possible with whatever string-like thing they want to use to identify their environment variable. They don't have to & anything.
* In steady-state code, _always_ take `&str` when your function chain only needs to have the string as long as the function chain exists, or when the clone is only occasional.
* In steady-state code, _probably_ take `impl Into<String>` when an owned `String` will be required to fulfill the function's contract (e.g., when your function requires filling out a protocol buffers object, which requires owned `String`s). The other choice is accepting an owned `String`, which certain microcontroller developers might prefer because there would be a couple fewer bytes of generated code. It's unlikely to be a consideration for momento's current target deployment scenarios.

Reply:
this is awesome, thanks so much. I will go back and apply this to all of the APIs when I pick up that String ticket, very soon :)